crates/graphql/src/schema.rs

```diff
@@ -204,58 +394,186 @@ impl<'a> From<models::Storefront<'a>> for Storefront {
-                }),
-            );
+        for listing in rows {
+            let listing = Listing::from(listing);
+
+            hash_map.insert(listing.address.clone(), Some(listing));
+        }
+
+        hash_map
+    }
+}
+
+pub struct StorefrontBatcher {
+    db_pool: Arc<Pool>,
+}
+
+#[async_trait]
+impl BatchFn<String, Option<Storefront>> for StorefrontBatcher {
+    async fn load(&mut self, keys: &[String]) -> HashMap<String, Option<Storefront>> {
+        let conn = self.db_pool.get().unwrap();
+        let mut hash_map = HashMap::new();
+
+        for key in keys {
+            hash_map.insert(key.clone(), None);
+        }
+
+        let columns = (
+            storefronts::owner_address,
+            storefronts::subdomain,
+            storefronts::title,
+            storefronts::description,
+            storefronts::favicon_url,
+            storefronts::logo_url,
+            storefronts::updated_at,
+            storefronts::banner_url,
+        );
+
+        let rows: Vec<models::Storefront> = storefronts::table
+            .select(columns)
+            .filter(storefronts::owner_address.eq(any(keys)))
+            .load(&conn)
+            .unwrap();
+
+        for storefront in rows {
+            let storefront = Storefront::from(storefront);
+
+            hash_map.insert(storefront.owner_address.clone(), Some(storefront));
         }
 
         hash_map
     }
 }
 
+pub struct ListingNftsBatcher {
+    db_pool: Arc<Pool>,
+}
+
+#[async_trait]
+impl BatchFn<String, Vec<Nft>> for ListingNftsBatcher {
+    async fn load(&mut self, keys: &[String]) -> HashMap<String, Vec<Nft>> {
+        let conn = self.db_pool.get().unwrap();
+        let mut hash_map = HashMap::new();
+
+        for key in keys {
+            hash_map.insert(key.clone(), Vec::new());
+        }
+
+        let rows: Vec<(String, models::Nft)> = listing_metadatas::table
+            .filter(listing_metadatas::listing_address.eq(any(keys)))
+            .inner_join(
+                metadatas::table.on(listing_metadatas::metadata_address.eq(metadatas::address)),
+            )
+            .inner_join(
+                metadata_jsons::table
+                    .on(listing_metadatas::metadata_address.eq(metadata_jsons::metadata_address)),
+            )
+            .select((
+                listing_metadatas::listing_address,
+                (
+                    metadatas::address,
+                    metadatas::name,
+                    metadata_jsons::description,
+                    metadata_jsons::image,
+                ),
+            ))
+            .load(&conn)
+            .unwrap();
+
+        rows.into_iter().fold(
+            hash_map,
+            |mut acc, (listing_address, nft): (String, models::Nft)| {
+                acc.entry(listing_address).and_modify(|nfts| {
+                    nfts.push(Nft::from(nft));
+                });
+
+                acc
+            },
+        )
+    }
```

Comment:
Is it possible to have this return a `Result` rather than calling `.unwrap()`?  It would be nice to avoid a panic.

Reply:
Yep. I'll follow up on this as part of this issue.

https://github.com/holaplex/metaplex-indexer/issues/52